### PR TITLE
feat(form-system-backend): Add functionality to create and edit custom lists

### PR DESCRIPTION
**Pull request by OlafurGeorg**
# ...

## What

Add functionality to the backend to create and edit custom lists

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/services/form-system/src/app/modules/forms/forms.service.ts
Comment (baering) [inline apps/services/form-system/src/app/modules/forms/forms.service.ts:329]: If all of these texts are supposed to be stored in code as strings it would be best to keep them together in translations files as objects and then reference them here

For example:

```javascript
// translations/form/create.ts

export const createFormTranslations: Record<string, LanguageType> = {
  premise: { is: 'Forsendur', en: Premises' },
}

// translations/form/index.ts
export * from './create'

// in code
import { createFormTranslations } from '../example/translations/form'

// ...
await this.sectionModel.create({
  name: createFormTranslations.premise
})
```

This way you will get typescript validation and you will keep changes to these files contained.

File: apps/services/form-system/src/app/modules/forms/forms.service.ts
Comment (baering) [inline apps/services/form-system/src/app/modules/forms/forms.service.ts:324]: Could also use:

```javascript
await this.secionModel.bulkCreate([
  // list of sections
])
```

File: apps/services/form-system/src/app/modules/forms/models/form.mapper.ts
Comment (baering) [inline apps/services/form-system/src/app/modules/forms/models/form.mapper.ts:20]: lodash/pick

File: apps/services/form-system/src/app/modules/applications/applications.controller.ts
Comment (baering) [inline apps/services/form-system/src/app/modules/applications/applications.controller.ts:8]: FYI you can version your controller like this:

```
@Controller({
  path: 'applications',
  version: ['1', VERSION_NEUTRAL],
})
```

`VERSION_NEUTRAL` indicates that this will work for any version passed in the request, or no version.

File: apps/services/form-system/src/app/modules/listItems/models/dto/updateListItem.dto.ts
Comment (baering) [inline apps/services/form-system/src/app/modules/listItems/models/dto/updateListItem.dto.ts:8]: You probably need `@ApiProperty{ type: LanguageType })` when it is not a base type and in this case also `@ApiPropertyOptional`

File: apps/services/form-system/src/app/modules/listItems/models/listItem.mapper.ts
Comment (baering) [inline apps/services/form-system/src/app/modules/listItems/models/listItem.mapper.ts:43]: Just be careful with object mutations as they can have unwanted side effects.

File: apps/services/form-system/src/app/modules/listItems/models/listItem.model.ts
Comment (baering) [inline apps/services/form-system/src/app/modules/listItems/models/listItem.model.ts:47]: `@ApiPropertyOptional`

File: package.json
Comment (baering) [inline package.json:452]: Remove this?

File: package.json
Comment (baering) [inline package.json:430]: What was required in this version?

**Review by baering — CHANGES_REQUESTED** [review]
I've finished reviewing about half the files. Posting the first batch of comments so you can review them. Will continue later.

Overall looking good. Noticed some things that would be good to refactor before merging, I’ve commented on some of them but some are repeating and I didn’t comment on each one.  Comments prefixed with `~` are considered minor.

I see the project is missing an infra definition so it will not get deployed anywhere. You should look into references on how other projects define their infrastructure setup. For example here is the infra setup for the [service portal](https://github.com/island-is/island.is/blob/main/apps/service-portal/infra/service-portal.ts) and here’s one for the [admin portal](https://github.com/island-is/island.is/blob/main/apps/portals/admin/infra/portals-admin.ts). Devops are [codeowners](https://github.com/island-is/island.is/blob/main/.github/CODEOWNERS#L371) of all infra setups so feel free to ping them for help.

Important things to look into next: 
- Scopes for nest controllers to ensure that access to endpoints is properly restricted based on roles and permissions. See [readme](https://github.com/island-is/island.is/blob/main/libs/auth-api-lib/seeders/README.md#creating-scopes) ([example PR for code reference](https://github.com/island-is/island.is/pull/15291/files))
- Validating that whoever is trying to make changes to or access any form/application etc is allowed to do it, both through scopes as mentioned above and manually in code by looking at national ids.
- Auditing, see [readme](https://github.com/island-is/island.is/blob/main/libs/nest/audit/README.md)
- Testing. As it boosts confidence in the code, catches bugs early, and ensures reliability. In the long run, it speeds up development by reducing manual testing and making future changes safer.

File: libs/portals/admin/form-system/src/utils/formatDate.tsx
Comment (baering) [inline libs/portals/admin/form-system/src/utils/formatDate.tsx:1]: We usually use `date-fns` for date formatting.

See here: [https://date-fns.org/v3.6.0/docs/format](https://date-fns.org/v3.6.0/docs/format).

**Review by baering — COMMENTED** [review]
Some more comments to review.

File: apps/services/form-system/docker-compose.yml
Comment (OlafurGeorg) [reply apps/services/form-system/docker-compose.yml:12]: 2/3, reyndi án árangurs að breyta username

File: apps/services/form-system/jest.config.ts
Comment (OlafurGeorg) [reply apps/services/form-system/jest.config.ts:10]: done

File: apps/services/form-system/migrations/20240605120243-create-organization.js
Comment (OlafurGeorg) [reply apps/services/form-system/migrations/20240605120243-create-organization.js:7]: done

File: apps/services/form-system/seeders/20240611110210-seed-field-type.js
Comment (OlafurGeorg) [reply apps/services/form-system/seeders/20240611110210-seed-field-type.js:7]: I get an error if I take out the id while seeding
ERROR: null value in column "id" of relation "organization" violates not-null constraint

File: apps/services/form-system/sequelize.config.js
Comment (OlafurGeorg) [reply apps/services/form-system/sequelize.config.js:6]: 2/3, reyndi árangurslaust að breyta username í þessum grunni

File: apps/services/form-system/src/app/enums/testimonyTypes.ts
Comment (OlafurGeorg) [reply apps/services/form-system/src/app/enums/testimonyTypes.ts:1]: renamed testimony to certification

File: apps/services/form-system/src/app/modules/sections/sections.controller.ts
Comment (OlafurGeorg) [reply apps/services/form-system/src/app/modules/sections/sections.controller.ts:36]: this method will soon be deleted as it has no real world use cases. It's being used for early development only

File: apps/services/form-system/src/app/modules/screens/screens.controller.ts
Comment (OlafurGeorg) [reply apps/services/form-system/src/app/modules/screens/screens.controller.ts:36]: this method will soon be deleted as it has no real world use cases. It's being used for early development only

File: apps/services/form-system/src/app/modules/fields/fields.controller.ts
Comment (OlafurGeorg) [reply apps/services/form-system/src/app/modules/fields/fields.controller.ts:36]: this method will soon be deleted as it has no real world use cases. It's being used for early development only

File: package.json
Comment (OlafurGeorg) [reply package.json:452]: removed

File: package.json
Comment (OlafurGeorg) [reply package.json:430]: changed back to 13.1.1

File: apps/services/form-system/migrations/20240614131225-create-testimony-type.js
Comment (OlafurGeorg) [reply apps/services/form-system/migrations/20240614131225-create-testimony-type.js:7]: I went with the word certification

File: apps/services/form-system/src/app/modules/listItems/models/listItem.mapper.ts
Comment (OlafurGeorg) [reply apps/services/form-system/src/app/modules/listItems/models/listItem.mapper.ts:43]: Is there a better way to do this in typescript?

File: libs/portals/admin/form-system/src/components/MainContent/components/InputContent/components/InputSettings/components/FIleUploadSettings.tsx
Comment (OlafurGeorg) [reply libs/portals/admin/form-system/src/components/MainContent/components/InputContent/components/InputSettings/components/FIleUploadSettings.tsx:92]: fixed on main

File: libs/portals/admin/form-system/src/lib/utils/fileTypes.ts
Comment (OlafurGeorg) [reply libs/portals/admin/form-system/src/lib/utils/fileTypes.ts:1]: Hyo-Sam will look into this

File: libs/portals/admin/form-system/src/components/Navbar/components/NavComponent/NavComponent.tsx
Comment (OlafurGeorg) [reply libs/portals/admin/form-system/src/components/Navbar/components/NavComponent/NavComponent.tsx:146]: This is fixed on main

File: libs/portals/admin/form-system/src/components/Navbar/components/NavComponent/NavComponent.tsx
Comment (OlafurGeorg) [reply libs/portals/admin/form-system/src/components/Navbar/components/NavComponent/NavComponent.tsx:137]: Hyo-Sam will look into this

File: apps/services/form-system/src/app/modules/forms/forms.service.ts
Comment (OlafurGeorg) [reply apps/services/form-system/src/app/modules/forms/forms.service.ts:324]: Three of the four sections fit into bulkcreate, but I need the id of the fourth one as foreign key for the screen

**Review by baering — APPROVED** [review]

**Comment by baering** [comment]
There's still a missing infra definition for the service, you can consult with the devops team on that.

**Review by Hyosam — APPROVED** [review]

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]
Infra LGTM

**Pull request merged**
